rootex/core/renderer/renderer.cpp

```diff
@@ -25,6 +25,37 @@ void Renderer::resetCurrentShader()
 	m_CurrentShader = nullptr;
 }
 
+void Renderer::bind(MaterialResourceFile* newMaterial, MaterialResourceFile* oldMaterial)
+{
+	if (newMaterial->getClassName() == "BasicMaterialResourceFile")
```

Comment:
I don't think this deserves to be here. This is a hot path. This function needs to do as less stuff as it can to get the best speed. Why has this been implemented like this?

Reply:
~string comparison~
dynamic_cast :heavy_check_mark: 

what we intend to do here is bind the oldMaterial's textures by default so that custom shaders can use them as well